src/hostname/hostnamed.c

```diff
@@ -1217,8 +1232,20 @@ static int method_set_hostname(sd_bus_message *m, void *userdata, sd_bus_error *
         /* We always go through with the procedure below without comparing to the current hostname, because
          * we might want to adjust hostname source information even if the actual hostname is unchanged. */
 
-        if (name && !hostname_is_valid(name, 0))
-                return sd_bus_error_setf(error, SD_BUS_ERROR_INVALID_ARGS, "Invalid hostname '%s'", name);
+        _cleanup_free_ char *substituted = NULL;
```

Comment:
method_set_static_hostname() does substitution after polkit check, but in this case before polkit check.
Maybe better to do the same here??